### PR TITLE
(feat) O3-2677: Improve the user menu

### PR DESCRIPTION
**Pull request by brandones**
# Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. Ensure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing#contributing-guidelines) label (such as `feat`, `fix`, or `chore`, among others). See existing PR titles for inspiration.

## For changes to apps

- [ ] My work conforms to the [**O3 Styleguide**](https://om.rs/styleguide) and [**design documentation**](https://om.rs/o3ui).

## If applicable

- [x] My work includes tests or is validated by existing tests.
- [ ] I have updated the [esm-framework mock](https://github.com/openmrs/openmrs-esm-core/blob/main/packages/framework/esm-framework/mock.tsx) to reflect any API changes I have made.

## Summary

Implementing the mock up @ibacher made in the ticket:

image

For the modal design I was winging it.

I also refactored the whole user menu. Removed all the extension state being passed around. Extension state is a point of fragility since it is not type-checked or anything like that. So now each user menu item is quite independent in how it handles data. Also refactored all the CSS, getting rid of most of it.

I preserved the existing translations for "change locale" where present. Changed the Spanish and French ones. I also used existing translations for "Confirm" for the "Apply" button.

## Screenshots

https://github.com/openmrs/openmrs-esm-core/assets/1031876/41224740-00b1-43fc-bb03-8cb9189dad40

## Related Issue

https://openmrs.atlassian.net/browse/O3-2677

### CODE REVIEW

File: packages/apps/esm-login-app/src/logout/logout.scss [hunk 1]
```diff
@@ -1,11 +1,11 @@
 @import '../root.scss';
 
-.logout {
-  padding-right: 0rem;
-  @extend .productiveHeading01;
-  width: 16rem;
+/* Repeat selector for specificity */
+button.logout.logout {
```
Comment (denniskigen) [inline packages/apps/esm-login-app/src/logout/logout.scss:4]: Should we maintain the original button width here so the user has a wider target to tap or click on?

File: packages/apps/esm-primary-navigation-app/src/components/change-language/change-language-link.extension.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import React, { useCallback } from 'react';
+import { Language } from '@carbon/react/icons';
+import { useTranslation } from 'react-i18next';
+import { showModal, useSession } from '@openmrs/esm-framework';
+import { Button } from '@carbon/react';
+import styles from './change-language-link.scss';
+import { SwitcherItem } from '@carbon/react';
```
Comment (denniskigen) [inline packages/apps/esm-primary-navigation-app/src/components/change-language/change-language-link.extension.tsx:7]: These `@carbon/react` imports should be consolidated.

**Review by denniskigen — APPROVED** [review]
Great execution, @brandones. Thanks! Left a few suggestions regarding Carbon imports.

**Comment by brandones** [comment]
Good catch & good call, thanks @denniskigen 

https://github.com/openmrs/openmrs-esm-core/assets/1031876/fb5ab81f-8ad4-4034-8438-efc5ef99ad11

**Pull request merged**